doc/ztp/ztp_hld.md

```diff
@@ -0,0 +1,166 @@
+# Zero Touch Provisioning (ZTP)
```

Comment:
Generally, I think many user ZTP flows will start from ONIE, not from a pre-installed SONiC image. I think this should be discussed a little. For instance, in an ONIE flow, the run-time image is usually installed by the ONIE installer, and not by the SONiC ZTP. So, in this flow, I think a ztp_image_url should be treated as optional, and lack of a valid one should not be treated as an error.